hazelcast/src/main/java/com/hazelcast/query/impl/predicates/FalsePredicate.java

```diff
@@ -73,7 +72,7 @@ public void readData(ObjectDataInput in) throws IOException {
 
     @Override
     public int getFactoryId() {
-        return PREDICATE_DS_FACTORY_ID;
+        return Factory.PREDICATE_DS.getDefaultFactoryId();
```

Comment:
> I don't see this as an improvement. The code has become a lot more verbose and this can lead to other performance problems.

I agree this isn't an improvement, but it's a necessary side effect of making them into an object.

Reply:
Then the design is problematic and should be fixed.

Reply:
@vbekiaris suggested that they could use a static `FACTORY_ID` variable in the class instead, which is a pattern already used elsewhere (e.g. `com.hazelcast.aggregation.impl.AggregatorDataSerializerHook.F_ID`). I've updated it to do this.